numpy/matrixlib/defmatrix.py

```diff
@@ -1024,8 +1024,8 @@ def _from_string(str, gdict, ldict):
             except KeyError:
                 try:
                     thismat = gdict[col]
-                except KeyError:
-                    raise KeyError("%s not found" % (col,))
+                except KeyError as e:
+                    raise KeyError("{!r} not found".format((col,)) from None
```

Comment:
```suggestion
                    raise KeyError("{!r} not found".format(col) from None
```
`(x,)` is part of the `%` formatting.

Perhaps more appropriate would be
```
```suggestion
                    raise NameError(col) from None
```